components/brave_wallet/common/test_utils.cc

```diff
@@ -35,4 +42,27 @@ mojom::NetworkInfo GetTestNetworkInfo2(const std::string& chain_id) {
           true};
 }
 
+scoped_refptr<value_store::TestValueStoreFactory> GetTestValueStoreFactory(
+    base::ScopedTempDir& temp_dir) {
+  CHECK(temp_dir.CreateUniqueTempDir());
+
+  base::FilePath test_data_dir;
+  CHECK(base::PathService::Get(base::DIR_SOURCE_ROOT, &test_data_dir));
+  base::FilePath src_db(
+      test_data_dir.AppendASCII("components/brave_wallet/test/data/"));
+  base::FilePath db_path = temp_dir.GetPath().AppendASCII("temp_db");
+  base::CopyDirectory(src_db, db_path, true);
+
+  return new value_store::TestValueStoreFactory(db_path);
```

Comment:
Do we still need this? As it seems tests are fine and there is not test database in repo.

Reply:
`value_store::TestValueStoreFactory()` will instantiate in-memory storage

Reply:
If a path is provided, it will open a database and we should run our test with real leveldb backed storage.
In tx state manager test, we do have test case like if the value is written into db or not.

Reply:
Ok for real database.
But what exactly is copied there?
```
  CHECK(base::PathService::Get(base::DIR_SOURCE_ROOT, &test_data_dir));
  base::FilePath src_db(
      test_data_dir.AppendASCII("components/brave_wallet/test/data/"));
  base::FilePath db_path = temp_dir.GetPath().AppendASCII("temp_db");
  base::CopyDirectory(src_db, db_path, true);
```

Reply:
Oh yes, we don't need that copy because we don't have any pre-created db to use. I forgot to remove it when I borrowed it from `components/value_store/value_store_frontend_unittest.cc`